### PR TITLE
GPT-1185-check-canvas-course-published-error

### PR DESCRIPTION
**Pull request by sastec17**
## 🤖 What does this PR do?
Throws custom error when canvas course is unpublished (`403 Forbidden` errpr)

## 📝 Any background context you want to provide?
Originally attempted to catch error in `umgpt` repo, but realized this library wasn't raising an exception at all - hence the direct changes here. 

Now, hooking up to this updated library from `umgpt`should throw the updated error without any additional changes to `umgpt` 

## 📑 Related Cards & Documents

- <https://um-its.atlassian.net/browse/GPT-1185>

## 🧑‍🔬 Testing steps
Connect to this version of canvas-langchain:
1. Change pyproject.toml `canvas-langchain` line to the following:
```
canvas-langchain = { git = "https://github.com/umich-its-ai/langchain-doc-canvas.git", branch = "GPT-1185-check-course-published"}
```
2. Update dependencies
```bash
poetry lock

# If python 3.12 is needed, run this before poetry lock
python3.12 -mvenv .venv  
source .venv/bin/activate
```
3. Update docker container via `docker-compose up --build --force-recreate`

If docker gets upset about finding LangChainKaltura - can add the following into DockerFile.local 
```
RUN pip install --no-cache-dir git+https://github.com/tl-its-umich-edu/langchain_kaltura@2025.03.01#egg=LangChainKaltura

```
4. Open an unpublished Canvas course in Maizey + attempt to reindex. Verify that you receive output seen in screenshot below.
5. Verify that indexing works normally for published canvas courses

## ✅ Checklist

- [ ] Changes satisfy card requirements
- [ ] Thoroughly tested, including edge cases and my own testing steps
- [ ] No new linting, formatting, or test issues
- [ ] Added/updated tests where applicable
- [ ] Checked for database migration conflicts
- [ ] Documented release steps in Jira
- [ ] Pointed to the correct base branch

## 📸 Screenshots/GIFs
This error should appear:

image

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -30,6 +30,7 @@
         "xlrd",
         "markdown",
         "pytz",
+        'LangChainKaltura @ git+https://github.com/tl-its-umich-edu/langchain_kaltura@2025.03.01#egg=LangChainKaltura'
```
Comment (jonathoy) [inline setup.py:33]: Is this something that should be included with this release or the next (with what I assume will have other Kaltura changes)?

**Review by jonathoy — COMMENTED** [review]
I need to test this still but leaving some code review feedback for now! As a reminder, we will need to coordinate this release with the one in UMGPT so that we don't cause any unintended effects.

File: setup.py (hunk 1 above)
Comment (sastec17) [reply setup.py:33]: I'm thinking it should be included with this release? I haven't made any changes to the LangChainKaltura library, but  this library fails to load in local testing without including the dependency in setup.py 

I'm not sure why this is working in the release without the LangChainKaltura library included here 🤔

Comment (jonathoy) [reply setup.py:33]: I think I see the problem. The Kaltura work was merged into main, which means that we can't really move your work up into a release until we separate that out (unless we are comfortable with it going in, which I don't have enough background to say if it is or not).

It feels like the changes would affect currently loader functionality in prod. Do you know enough about this to confirm? This might need to be something we discuss as a larger group to figure out if we need to untangle the code.
